UnitTest failures on main

Running on main I'm seeing test failures.

```
git log
commit 61fc8785ab8adb4cec5f1135d278a9e8895f327c (HEAD, upstream/main, upstream/HEAD)
Author: Sebastian Tiedtke <sebastiantiedtke@gmail.com>
Date:   Thu May 23 16:41:17 2024 -0400

    Is python3 more likely than python? (#586)
    
    `python` is an alias for `python3` on my BMP which makes the test fail.
```

Here are the test failures

```
 logger.go:146: 2024-05-23T23:09:03.531Z	INFO	stream canceled after the process finished; ignoring	{"id": "01HYKVE2NV87J6ZQTMY69E6BYT"}
    --- FAIL: TestRunnerServiceServerExecute_WithInput/SimulateCtrlC (1.52s)
        service_execute_test.go:643: 
            	Error Trace:	/Users/jlewi/git_runme/internal/runnerv2service/service_execute_test.go:643
            	Error:      	"rpc error: code = Unknown desc = exit status 1" does not contain "exit status 130"
            	Test:       	TestRunnerServiceServerExecute_WithInput/SimulateCtrlC
        service_execute_test.go:644: 
            	Error Trace:	/Users/jlewi/git_runme/internal/runnerv2service/service_execute_test.go:644
            	Error:      	Not equal: 
            	            	expected: 130
            	            	actual  : 1
            	Test:       	TestRunnerServiceServerExecute_WithInput/SimulateCtrlC
```

```
    --- FAIL: TestRunnerServiceServerExecute_WithInput/ContinuousInput (0.04s)
        service_execute_test.go:600: 
            	Error Trace:	/Users/jlewi/git_runme/internal/runnerv2service/service_execute_test.go:600
            	Error:      	Not equal: 
            	            	expected: "a\r\nb\r\nc\r\nd\r\nA\r\nB\r\nC\r\nD\r\n"
            	            	actual  : "a\r\nb\r\nA\r\nB\r\nc\r\nC\r\nd\r\nD\r\n"
            	            	
            	            	Diff:
            	            	--- Expected
            	            	+++ Actual
            	            	@@ -2,7 +2,7 @@
            	            	 b
            	            	-c
            	            	-d
            	            	 A
            	            	 B
            	            	+c
            	            	 C
            	            	+d
            	            	 D
            	Test:       	TestRunnerServiceServerExecute_WithInput/ContinuousInput
```

```
2024-05-23T23:09:03.740Z	DEBUG	runner/command.go:404	finished copying from pty to stdout	{"count": 32}
--- FAIL: Test_command (0.00s)
    --- FAIL: Test_command/JavaScript (0.03s)
        command_test.go:107: 
            	Error Trace:	/Users/jlewi/git_runme/internal/runner/command_test.go:107
            	Error:      	Not equal: 
            	            	expected: ""
            	            	actual  : "(node:24760) ExperimentalWarning: The Node.js specifier resolution flag is experimental. It could change or be removed at any time.\n(Use `node --trace-warnings ...` to show where the warning was created)\n"
            	            	
            	            	Diff:
            	            	--- Expected
            	            	+++ Actual
            	            	@@ -1 +1,3 @@
            	            	+(node:24760) ExperimentalWarning: The Node.js specifier resolution flag is experimental. It could change or be removed at any time.
            	            	+(Use `node --trace-warnings ...` to show where the warning was created)
            	            	 
            	Test:       	Test_command/JavaScript
    --- FAIL: Test_command/JavaScriptEnv (0.03s)
        command_test.go:135: 
            	Error Trace:	/Users/jlewi/git_runme/internal/runner/command_test.go:135
            	Error:      	Not equal: 
            	            	expected: ""
            	            	actual  : "(node:24765) ExperimentalWarning: The Node.js specifier resolution flag is experimental. It could change or be removed at any time.\n(Use `node --trace-warnings ...` to show where the warning was created)\n"
            	            	
            	            	Diff:
            	            	--- Expected
            	            	+++ Actual
            	            	@@ -1 +1,3 @@
            	            	+(node:24765) ExperimentalWarning: The Node.js specifier resolution flag is experimental. It could change or be removed at any time.
            	            	+(Use `node --trace-warnings ...` to show where the warning was created)
            	            	 
            	Test:       	Test_command/JavaScriptEnv
FAIL
FAIL	github.com/stateful/runme/v3/internal/runner	4.624s
ok  	github.com/stateful/runme/v3/internal/runner/client	0.255s
--- FAIL: TestRunnerServiceServerExecute_WithInput (1.61s)
```

Tests appear to have passed on main at this commit #586 .

Are others seeing test failures when running locally or is it just me?

😮 this is actually a bug in how the vscode terminal connects back to the kernel server running inside of vscode. The env vars seem to be persistent when reloaded/revived. I can repro the issue. Attempting a fix here: https://github.com/stateful/vscode-runme/pull/1379

```
 logger.go:146: 2024-05-23T23:09:03.531Z	INFO	stream canceled after the process finished; ignoring	{"id": "01HYKVE2NV87J6ZQTMY69E6BYT"}
    --- FAIL: TestRunnerServiceServerExecute_WithInput/SimulateCtrlC (1.52s)
        service_execute_test.go:643: 
            	Error Trace:	/Users/jlewi/git_runme/internal/runnerv2service/service_execute_test.go:643
            	Error:      	"rpc error: code = Unknown desc = exit status 1" does not contain "exit status 130"
            	Test:       	TestRunnerServiceServerExecute_WithInput/SimulateCtrlC
        service_execute_test.go:644: 
            	Error Trace:	/Users/jlewi/git_runme/internal/runnerv2service/service_execute_test.go:644
            	Error:      	Not equal: 
            	            	expected: 130
            	            	actual  : 1
            	Test:       	TestRunnerServiceServerExecute_WithInput/SimulateCtrlC
```

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -57,12 +57,6 @@ jobs:
         run: |
           go build -o runme main.go
           ./runme --version
-      - name: Test All
-        run: |
-          export SHELL=/bin/bash
-          export TZ=UTC
-          make test
-        if: ${{ matrix.os == 'ubuntu-latest' }}
       - name: Test
         run: |
           export SHELL=/bin/bash
@@ -70,10 +64,7 @@ jobs:
           TAGS="test_with_docker" make test/coverage
           make test/coverage/func
         if: ${{ matrix.os == 'ubuntu-latest' }}
-      # make test should run on all os's because
-      # on windows only a subset of the tests run.
       - name: Test
-        # N.B. Set-Timezone is a windows command
         run: |
           Set-Timezone -Id "UTC" -PassThru
           make test

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -61,7 +61,7 @@ jobs:
         run: |
           export SHELL=/bin/bash
           export TZ=UTC
-          make test          
+          make test
         if: ${{ matrix.os == 'ubuntu-latest' }}
       - name: Test
         run: |
@@ -69,15 +69,15 @@ jobs:
           export TZ=UTC
           TAGS="test_with_docker" make test/coverage
           make test/coverage/func
-        if: ${{ matrix.os == 'ubuntu-latest' }}      
+        if: ${{ matrix.os == 'ubuntu-latest' }}
       # make test should run on all os's because
       # on windows only a subset of the tests run.
       - name: Test
-        # N.B. Set-Timezone is a windows command 
+        # N.B. Set-Timezone is a windows command
         run: |
           Set-Timezone -Id "UTC" -PassThru
           make test
-        if: ${{ matrix.os == 'windows-latest' }}        
+        if: ${{ matrix.os == 'windows-latest' }}
       - name: Upload coverage
         uses: actions/upload-artifact@v4
         if: ${{github.actor != 'dependabot[bot]' && matrix.os == 'ubuntu-latest'}}

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -134,6 +134,10 @@ jobs:
 
       - name: SonarCloud Scan
         uses: SonarSource/sonarcloud-github-action@v2.1.1
+        # Skip this test if the PR is created from a fork.
+        # If its created from a fork the PR won't be able to fetch the secrets so
+        # the step will fail.
+        if: github.event.pull_request.head.repo.full_name == github.repository
         env:
           GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }} # Needed to get PR information, if any
           SONAR_TOKEN: ${{ secrets.SONAR_TOKEN }}

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -64,11 +64,12 @@ jobs:
           TAGS="test_with_docker" make test/coverage
           make test/coverage/func
         if: ${{ matrix.os == 'ubuntu-latest' }}
+      # make test should run on all os's because
+      # on windows only a subset of the tests run.
       - name: Test
         run: |
           Set-Timezone -Id "UTC" -PassThru
-          make test
-        if: ${{ matrix.os == 'windows-latest' }}
+          make test        
       - name: Upload coverage
         uses: actions/upload-artifact@v4
         if: ${{github.actor != 'dependabot[bot]' && matrix.os == 'ubuntu-latest'}}

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -57,19 +57,27 @@ jobs:
         run: |
           go build -o runme main.go
           ./runme --version
+      - name: Test All
+        run: |
+          export SHELL=/bin/bash
+          export TZ=UTC
+          make test          
+        if: ${{ matrix.os == 'ubuntu-latest' }}
       - name: Test
         run: |
           export SHELL=/bin/bash
           export TZ=UTC
           TAGS="test_with_docker" make test/coverage
           make test/coverage/func
-        if: ${{ matrix.os == 'ubuntu-latest' }}
+        if: ${{ matrix.os == 'ubuntu-latest' }}      
       # make test should run on all os's because
       # on windows only a subset of the tests run.
       - name: Test
+        # N.B. Set-Timezone is a windows command 
         run: |
           Set-Timezone -Id "UTC" -PassThru
-          make test        
+          make test
+        if: ${{ matrix.os == 'windows-latest' }}        
       - name: Upload coverage
         uses: actions/upload-artifact@v4
         if: ${{github.actor != 'dependabot[bot]' && matrix.os == 'ubuntu-latest'}}

--- a/internal/runner/service_test.go
+++ b/internal/runner/service_test.go
@@ -347,7 +347,7 @@ func Test_runnerService(t *testing.T) {
 	})
 
 	t.Run("Input", func(t *testing.T) {
-		if os.Getenv("RUN_FLAKY") == "" {
+		if skip, err := strconv.ParseBool(os.Getenv("SKIP_FLAKY")); err == nil && skip {
 			t.Skip("skipping flaky test")
 		}
 		t.Parallel()

--- a/internal/runner/service_test.go
+++ b/internal/runner/service_test.go
@@ -11,6 +11,7 @@ import (
 	"os"
 	"os/exec"
 	"runtime"
+	"strconv"
 	"strings"
 	"syscall"
 	"testing"

--- a/internal/runner/service_test.go
+++ b/internal/runner/service_test.go
@@ -347,6 +347,9 @@ func Test_runnerService(t *testing.T) {
 	})
 
 	t.Run("Input", func(t *testing.T) {
+		if os.Getenv("RUN_FLAKY") == "" {
+			t.Skip("skipping flaky test")
+		}
 		t.Parallel()
 
 		stream, err := client.Execute(context.Background())

